File: 43165.cpp
#include <string>
#include <vector>

using namespace std;

int dfs(vector<int> numbers, int size, int target, int currentValue, int index){
    if(index == size)
        if(currentValue == target)  return 1;
        else return 0;
        
    return  dfs(numbers, size, target, currentValue + numbers.at(index), index + 1) + 
            dfs(numbers, size, target, currentValue - numbers.at(index), index + 1);
}

int solution(vector<int> numbers, int target) {
    int size = numbers.size();
    return dfs(numbers, size, target, 0, 0);
}

File: 43165.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    return DFS(numbers, target, 0, 0)
}

func DFS(_ numbers: [Int], _ target: Int, _ result: Int, _ index: Int) -> Int {
    if index >= numbers.count {
        return result == target ? 1 : 0
    } else {
        return DFS(numbers, target, result + numbers[index], index + 1) + DFS(numbers, target, result - numbers[index], index + 1)
    }
}
